core/common/src/main/java/alluxio/exception/NoMasterException.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.exception;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * The exception thrown when a master does not exist, i.e. the id of the master cannot be found.
+ */
+@ThreadSafe
+public class NoMasterException extends AlluxioException {
```

Comment:
MetaMasterMasterServiceHandler
```
  public RegisterMasterTResponse registerMaster(final long masterId,
      RegisterMasterTOptions options) throws AlluxioTException {
    return RpcUtils.call(LOG, new RpcUtils.RpcCallable<RegisterMasterTResponse>() {
      @Override
      public RegisterMasterTResponse call() throws AlluxioException {
        mMasterProcess.masterRegister(masterId, options);
        return new RegisterMasterTResponse();
      }
```
Because of the ServiceHandler, I can only throw AlluxioException here. NoMasterException and NoWorkerException(used by workerRegister) is under alluxio.exception package, so I can use them here.  Those exceptions under alluxio.exception.status packages cannot be throw here.

Reply:
You can use `RpcCallableThrowsIOException`, see https://github.com/Alluxio/alluxio/blob/master/core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientServiceHandler.java#L138